runtime/parser2/expression.go

```diff
@@ -598,7 +598,7 @@ func defineLessThanOrTypeArgumentsExpression() {
 				defer func() {
 					err := recover()
 					// Fatal errors should abort parsing
```

Comment:
Maybe update the comment above.

Maybe this shouldn't check memory errors specifically, but e.g. non-`ParseError`s?

Reply:
Ideally Yes. But many parser errors are thrown with `fmt.Errorf()`, so they won't get caught as `ParseError`. We need to throw specific errors (i.e: replace `fmt.Errorf()` with a known `ParseError` similar to what we did in this PR for interpreter errors).

I was hoping to do that in a separate PR, to avoid this becoming too large.

Reply:
Ah, that makes sense! Yes, we can/should do that in a separate PR.
BTW, I had discussed the errors in the parser with @robert-e-davidson3 before, who had asked about it:

> when a syntax problem is encountered there are currently four (maybe more?) “kinds” how that is reported: 
> - `panic` with a string, e.g. `panic("expected comma, got start of parameter")`
> - `panic` with a “plain” error, e.g. `panic(fmt.Errorf("expected parameter or end of parameter list, got %s", p.current.Type))`
> - `parser.report` with a “plain” error, e.g. `report(fmt.Errorf("missing start of string literal: expected '\"'"))`
> - `parser.report` with a `SyntaxError`
> `panic`s abort the parse immediately, and if the value that’s recovered (e.g. string or plain error) is not a `SyntaxError`, it is automatically wrapped into one at the current parse location

Maybe we can put this into a separate issue